experimental/oak_baremetal_runtime/src/lib.rs

```diff
@@ -33,53 +33,20 @@ mod wasm;
 use crate::{
     logger::StandaloneLogger,
     remote_attestation::{AttestationHandler, AttestationSessionHandler},
-    schema::TrustedRuntime,
 };
 use alloc::{boxed::Box, sync::Arc};
 use anyhow::Context;
-use oak_baremetal_communication_channel::{
-    server::{message_from_response_and_id, ServerChannelHandle},
-    Channel,
-};
 use oak_functions_abi::Request;
 use oak_functions_lookup::LookupDataManager;
-use oak_idl::Handler;
 use oak_remote_attestation::handshaker::{
     AttestationBehavior, AttestationGenerator, AttestationVerifier,
 };
 use oak_remote_attestation_sessions::SessionId;
 
-/// Sole entrypoint to the runtime. Starts the runtime and listens for requests
-/// on the provided channel.
-pub fn start<G: 'static + AttestationGenerator, V: 'static + AttestationVerifier>(
-    channel: Box<dyn Channel>,
-    attestation_behavior: AttestationBehavior<G, V>,
-) -> anyhow::Result<!> {
-    let mut invocation_handler = RuntimeImplementation {
-        initialization_state: InitializationState::Uninitialized(Some(attestation_behavior)),
-        lookup_data_manager: Arc::new(LookupDataManager::new_empty(StandaloneLogger::default())),
-    }
-    .serve();
-    let channel_handle = &mut ServerChannelHandle::new(channel);
-    loop {
-        let request_message = channel_handle
-            .read_request()
-            .context("couldn't receive message")?;
-        let request_message_invocation_id = request_message.invocation_id;
-        let response = invocation_handler.invoke(request_message.into());
-        // For now all messages are sent in sequence, hence the id of the next
-        // response always matches that of the preceeding request.
-        // TODO(#2848): Allow messages to be sent and received out of order.
```

Comment:
This TODO seems to have gone missing from the resulting code, even though I think it is still relevant.

Reply:
Good catch! I think it's fine to remove it from this spot, since it deals just with responding requests. We probably want the single-threaded runtime to respond to request in the order the come in. 

The todo still exists in the launcher, which deals the order in which requests are sent (and which atm waits for the response before sending a new requests). See https://github.com/project-oak/oak/blob/071dd8847569aab186dd269fe6da181aee872df1/experimental/oak_baremetal_launcher/src/channel.rs#L45